app/peptide/payloadstore/payloadstore.go

```diff
@@ -1,70 +1,100 @@
 package payloadstore
 
 import (
-	"sync"
+	"fmt"
 
 	"github.com/ethereum/go-ethereum/beacon/engine"
 	"github.com/polymerdao/monomer"
 )
 
-type PayloadStore interface {
-	Add(payload *monomer.Payload)
-	Get(id engine.PayloadID) (*monomer.Payload, bool)
-	Current() *monomer.Payload
-	RollbackToHeight(height int64) error
-}
-
-type pstore struct {
-	mutex    sync.Mutex
+type Store struct {
 	payloads map[engine.PayloadID]*monomer.Payload
 	heights  map[int64]engine.PayloadID
 	current  *monomer.Payload
 }
 
-var _ PayloadStore = (*pstore)(nil)
-
-func NewPayloadStore() PayloadStore {
-	return &pstore{
-		mutex:    sync.Mutex{},
+func NewPayloadStore() *Store {
+	return &Store{
 		payloads: make(map[engine.PayloadID]*monomer.Payload),
 		heights:  make(map[int64]engine.PayloadID),
 	}
 }
 
-func (p *pstore) Add(payload *monomer.Payload) {
+// Add adds a payload to the store and updates the current payload
+// if the new payload has a greater height.
+func (p *Store) Add(payload *monomer.Payload) {
 	id := payload.ID()
-	p.mutex.Lock()
-	defer p.mutex.Unlock()
 	if _, ok := p.payloads[*id]; !ok {
 		p.heights[payload.Height] = *id
 		p.payloads[*id] = payload
-		p.current = payload
+
+		if (p.current == nil) || (payload.Height > p.current.Height) {
```

Comment:
The current payload isn't necessarily the payload with the greatest height. It is the payload that will become the next block.